substrate/frame/support/src/traits/tokens/fungible/regular.rs

```diff
@@ -181,8 +181,9 @@ pub trait Unbalanced<AccountId>: Inspect<AccountId> {
 	) -> Result<Self::Balance, DispatchError> {
 		let old_balance = Self::balance(who);
 		let free = Self::reducible_balance(who, preservation, force);
-		if let BestEffort = precision {
-			amount = amount.min(free);
+		match precision {
+			BestEffort => amount = amount.min(free),
+			Exact => ensure!(free >= amount, TokenError::FundsUnavailable),
 		}
 		let new_balance = old_balance.checked_sub(&amount).ok_or(TokenError::FundsUnavailable)?;
```

Comment:
We shouldn't need this check anymore, as `amount` <= `free` <= `old_balance` (by definition `reducible_balance` must be always <= than `balance`)
https://github.com/paritytech/polkadot-sdk/blob/676bacd7299826fb0ea88055d2108f3c6a734291/substrate/frame/support/src/traits/tokens/fungible/regu
lar.rs#L80-L90
Same applies to `fungibles`